### PR TITLE
More stringent range testing

### PR DESCRIPTION
**Pull request by timholy**
Turns a `@test_skip` into a `@test`. This may have been passing since #23194 but was never turned into a requirement. This change removes the single largest contribution to "total broken tests" in the whole test suite.

### CODE REVIEW

**Comment by LilithHafner** [comment]
```
Error in testset ranges:
--
  | Test Failed at C:\buildkite-agent\builds\win2k22-amdci6-4\julialang\julia-master\julia-3d18b7b681\share\julia\test\ranges.jl:890
  | Expression: stop ≈ last(r)
  | Evaluated: -6.67572f-5 ≈ -6.681681f-5
```

**Comment by timholy** [comment]
Yeah, there seem to be a few edge cases left. Would could allow a small failure rate, or just close this. Thoughts?

**Comment by simonbyrne** [comment]
Why is it failing only on some platforms? Is it not deterministic?

**Comment by simonbyrne** [comment]
An example case I found:
```julia
julia> (start, delta, stop)  = -2.8345144f0, 0.9447948f0, -0.00012993813f0
(-2.8345144f0, 0.9447948f0, -0.00012993813f0)

julia> r = start:delta:stop
-2.8345144f0:0.9447948f0:-0.00013005733f0

julia> last(r) ≈ stop
false
```

The problem here is that
```julia
julia> start + 3*delta  # intermediate values in Float32
-0.00012993813f0

julia> Float32(Float64(start) + 3*Float64(delta)) # intermediate values in higher precision
-0.00013005733f0
```

This particular case uses the [fallback branch](https://github.com/JuliaLang/julia/blob/b48104d5f29fcfd477c72b1baa22145297c08bb6/base/twiceprecision.jl#L421-L434), since the rational part fails:
```julia
julia> num, den = Base.rat(delta)
(599, 634)

julia> Float32(num/den)
0.94479495f0
```
The question here is what criteria we should use for determining `delta` and `stop`

File: test/ranges.jl [hunk 1]
```diff
@@ -887,7 +887,7 @@ function range_fuzztests(::Type{T}, niter, nrange) where {T}
         @test m == length(r)
         @test strt == first(r)
         @test Δ == step(r)
-        @test_skip stop ≈ last(r)
+        @test stop ≈ last(r)
```
Comment (simonbyrne) [inline test/ranges.jl:890]:
```suggestion
        @test stop ≈ last(r) atol = eps((n-1)*Δ) + eps(stop)
```

**Review by oscardssmith — APPROVED** [review]

File: test/ranges.jl [hunk 2]
```diff
@@ -887,7 +887,7 @@ function range_fuzztests(::Type{T}, niter, nrange) where {T}
         @test m == length(r)
         @test strt == first(r)
         @test Δ == step(r)
-        @test_skip stop ≈ last(r)
+        @test stop ≈ last(r) atol = eps((n-1)*Δ) + eps(stop)
```
Comment (IanButterworth) [inline test/ranges.jl:890]: @simonbyrne your explanation in the PR seems like a good comment
```suggestion
        @test stop ≈ last(r) atol = eps((n-1)*Δ) + eps(stop) # the tolerance needs to take into account the intermediate rounding when computing stop
```

**Comment by timholy** [comment]
LGTM, thanks @simonbyrne

**Comment by vtjnash** [comment]
Seen on CI:
```
The global RNG seed was 0xf5fc6174bf915507070615bf2c22eaf4.
Error in testset ranges:
Test Failed at /Users/julia/.julia/scratchspaces/a66863c6-20e8-4ff4-8a62-49f30b1f605e/agent-cache/default-macmini-aarch64-4.0/build/default-macmini-aarch64-4-0/julialang/julia-master/julia-d7b2062924/share/julia/test/ranges.jl:890
  Expression: ≈(stop, last(r), atol = eps((n - 1) * Δ) + eps(stop))
   Evaluated: -1.6152623f0 ≈ -1.6152625f0 (atol=1.7881393e-7)
```

**Pull request merged**